src/main/kotlin/fi/metatavu/metaform/server/controllers/UsersController.kt

```diff
@@ -0,0 +1,109 @@
+package fi.metatavu.metaform.server.controllers
+
+import fi.metatavu.metaform.api.spec.model.User
+import fi.metatavu.metaform.api.spec.model.UserFederatedIdentity
+import fi.metatavu.metaform.api.spec.model.UserFederationSource
+import fi.metatavu.metaform.keycloak.client.models.UserRepresentation
+import java.util.*
+import javax.enterprise.context.ApplicationScoped
+import javax.inject.Inject
+
+/**
+ * Controller for User objects
+ */
+@ApplicationScoped
+class UsersController {
+
+    @Inject
+    lateinit var cardAuthKeycloakController: CardAuthKeycloakController
+
+    @Inject
+    lateinit var metaformKeycloakController: MetaformKeycloakController
+
+    /**
+     * Creates User in Metaform Keycloak
+     *
+     * @param user User
+     * @return UserRepresentation
+     */
+    fun createUser(user: User): UserRepresentation {
+        val createdUser = metaformKeycloakController.createUser(user)
+
+        if (!user.federatedIdentities.isNullOrEmpty()) {
+            return metaformKeycloakController.createUserFederatedIdentity(
+                userId = UUID.fromString(createdUser.id),
+                userFederatedIdentity = user.federatedIdentities[0]
+            )
+        }
+
+        return createdUser
+    }
+
+    /**
+     * Updates User in Metaform Keycloak
+     *
+     * @param userId userId
+     * @param user User
+     * @return UserRepresentation
+     */
+    fun updateUser(userId: UUID, user: User): UserRepresentation? {
+        if (user.displayName.isNullOrEmpty()) {
+            return null
+        }
+
+        val splittedName = user.displayName.split(" ")
+        if (splittedName.size == 3) {
```

Comment:
Umm... could you explain the logic behind this?

Reply:
Rather not..... Removed.